### PR TITLE
chore(dev): Ignore directories created by check-component-features and clean them

### PR DESCRIPTION
**Pull request by spencergilbert**
Noticed these two issues when running the make targets locally

### CODE REVIEW

**Review by neuronull — APPROVED** [review]

File: .gitignore [hunk 1]
```diff
@@ -12,8 +12,9 @@ bench_output.txt
 sample.log
 scripts/package-lock.json
 target
+# Generated by `make check-component-features`
+target-*/
```
Comment (StephenWakely) [inline .gitignore:16]: Why ignore `target-*`, but delete `target-feature-*` below?

Wouldn't it be better if they were both `target-feature-*`?

Comment (spencergilbert) [reply .gitignore:16]: Yeah, changed it on the Make side and forgot to change here.

**Comment by spencergilbert** [comment]
Holding off merging, pending @tobz exploration of other fixes.

**Comment by spencergilbert** [comment]
Replaced by: https://github.com/vectordotdev/vector/pull/13996

**Pull request closed without merging**